### ISSUE TITLE
make home work properly

### ISSUE THREAD

**Issue by Raynes** [Raynes/fs#12]
https://github.com/Raynes/fs/commit/1b5e18ee9bc83a5bcaaddbf7f9667834b0cf7f87#commitcomment-796025

Blah. This can be fixed, but boy will it be ugly.

**Comment by xpe** [comment]
@Raynes Does this problem remain? Perhaps it would help to add small number of examples to reproduce the problems? (I'm about to start digging into ~ expansion with Java interop, so I can lend another set of eyes.)

**Comment by Raynes** [comment]
Yes. I'm not sure what I could add as an example. The problem is that fs doesn't actually know where `~<user>` leads, because it doesn't necessarily have to be `/home/<user>`.

**Comment by xpe** [comment]
@Raynes I don't see any examples. When I dig through the link above, I don't see a clear statement of the problem.

No, at the risk of having no clarity on what the actual problem is, why is this hard? Why not just lookup the $HOME environment variable and convert ~ to it?

**Comment by Raynes** [comment]
Please read the linked thread. The problem is clearly defined.

`~<user>/foo`

We don't currently have a way to tell where <user>'s home directory actually is. `$HOME` is not remotely cross platform, let alone even reliably set in every environment. Not sure how much more I can explain :P

**Comment by Raynes** [comment]
It's not necessarily *hard* either, I just haven't bothered.

**Comment by xpe** [comment]
$HOME works most of the time according to http://stackoverflow.com/questions/585534. Surely there is a respectable Java library that has already solved this?

**Comment by Raynes** [comment]
Fantastic question. Is there?
